File: proc-bm-wrapped+return.js

if (typeof(print) === 'undefined') {
    print = console.log
}

function e(x) { return x; }
// convert to and from a string
//function e(x) { return ("" + x) - 0 }

var f = 0;
f = e;

function g2(x) {
  return [f(x), function(y) { return y; } ];
}

print("wrapped+return");
function benchmark(fun) {
  var milliseconds1 = Date.now();
  var result = fun();
  var milliseconds2 = Date.now();
  var timing = (milliseconds2 - milliseconds1).toString();
  if (timing.indexOf(".") === -1) {
    timing = timing + ".0";
  }
  print("RESULT-cpu: " + timing);
  print("RESULT-total: " + timing);
}


benchmark(function() {
  for (i = 0; i < 10000000; i++) {
    var l = g2(i);
    l[1](l[0]);
  }
})

File: proc-bm-wrapped+return.py

import time

def e(x):
    return x

f = 0
f = e

def g2(x):
    return f(x), lambda y: y

print "wrapped+return"

start = time.clock() * 1000
for i in xrange(10000000):
    l, r = g2(i)
    r(l)
delta = time.clock() * 1000 - start

print "RESULT-cpu: %s\nRESULT-total: %s" % (delta, delta)
